### DISCUSSION TITLE
Text-based search in vitrivr-ng

### DISCUSSION THREAD

**Discussion by newtonsp** [ppanopticon/vitrivr-ng#67] [Q&A] [answered]
Dear devs,

I used an ancient version of vitrivr (probably the one associated with cineast 2.0.0) until recently. By the way, it was very helpful in the video retrieval prototype of our research group. In that version, I could do text-based search using an option available in the vitrivr-ng web page. After choosing text search, a window opened and I chose "metadata" to insert text. This action called methods that I could recognize and integrate with our prototype.

Then few months ago I started to migrate to a newer version (3.0.1). I was able to run the example provided in vitrivr page for all the stack. Then I tried to integrate the stack with our prototype. Cottontail and cineast are already OK, but vitrivr-ng is not. In particular, I can see the vitrivr-ng page, with an updated GUI and options like image and 3d based search. However, I couldn't find nor activate text-based search. 

I don't understand what is wrong, as the config.json file is already activating text-based (I attached it if you wanna have a look, with a different extension to make upload possible). It has a few changes from the example, based on the settings that worked for me in the old stack. And I couldn't solve the issue alone, after reading the Wiki pages.

Could you help me, for example, by... 

1. saying what I should do to activate text-based search or a similar alternative, or
2. giving me the first steps to be able to use tags as if they were the old text-based search, or
3. indicating me which vitrivr-ng source files I should change to include a text field and call/trigger cineast methods for retrieval, like the ones from the ContinousQueryDispatcher class, or
4. any other idea?

My knowledge in TypeScript is minimal, so if there's more than one alternative, I prefer the one that depends less on that language.

Thank you so much
Sincerely
[config.json.txt](https://github.com/vitrivr/vitrivr-ng/files/5968807/config.json.txt)

**Comment by lucaro** [comment, marked as answer]
Hey,

thanks for your message. I just looked at your config and the problem appears to be as simple as a missing comma on line 11, which makes the json invalid and hence not parsable. When the config cannot be loaded, vitrivr-ng defaults to an internal config. The error is however only shown in the javascript console, maybe we need to show it more explicitly in the UI itself.

Please let us know if that already solves your issue.

**Comment by newtonsp** [comment]
Dear Luca,

Thank you so much! I haven't noted my silly error...

Now I found text-based search and the cineast methods that process it.

Sincerely


Newton Spolaôr

Professor na Universidade Estadual do Oeste do Paraná (UNIOESTE)
Pesquisador no Laboratório de Bioinformática (LABI)
Doutorado na Universidade de São Paulo (USP)


Em sex., 12 de fev. de 2021 às 04:35, Luca Rossetto <
notifications@github.com> escreveu:
